File: sample-conference-videochat/sample-conference-videochat/Components/SharingViewController/SharingScreenCapture.m
//
//  QBRTCScreenCaptuerer.m
//  QuickbloxWebRTC
//
//  Created by Injoit on 2/25/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

#import "SharingScreenCapture.h"

@interface SharingScreenCapture()

@property (strong, nonatomic) VideoFormat *videoFormat;

@end

@implementation SharingScreenCapture

- (instancetype)initWithVideoFormat:(VideoFormat *)videoFormat {
    
    self = [super init];
    if (self) {
        
        _videoFormat = videoFormat;
    }
    
    return self;
}


#pragma mark - <QBRTCVideoCapture>

- (void)didSetToVideoTrack:(QBRTCLocalVideoTrack *)videoTrack {
    [super didSetToVideoTrack:videoTrack];
    
    [self adaptOutputFormatToWidth:self.videoFormat.width height:self.videoFormat.height fps:self.videoFormat.fps];
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/SharingViewController/SharingScreenCapture.swift
//
//  SharingScreenCapture.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 30.06.2021.
//  Copyright © 2021 QuickBlox. All rights reserved.
//

import UIKit
import QuickbloxWebRTC

struct VideoFormat {
    var width: UInt
    var height: UInt
    var fps: UInt
}

class SharingScreenCapture: QBRTCVideoCapture {
    //MARK: - Properties
    var sharingFormat: VideoFormat!
    //MARK: - Life Cycle
    init(sharingFormat: VideoFormat) {
        super.init()
        
        self.sharingFormat = sharingFormat
    }
    
    // MARK: - <QBRTCVideoCapture>
    override func didSet(to videoTrack: QBRTCLocalVideoTrack?) {
        super.didSet(to: videoTrack)
        
        adaptOutputFormat(toWidth: sharingFormat.width, height: sharingFormat.height, fps: sharingFormat.fps)
    }
}
